public/jsx/app.jsx

```diff
@@ -25,10 +27,23 @@ var App = React.createClass({
     var productId = router.getCurrentQuery().product;
 
     return {
+      error: AppStore.getError(),
       productId: productId,
     };
   },
 
+  componentDidMount: function() {
+    AppStore.addErrorListener(this.onError);
```

Comment:
we have a hard dependency between the component and store here. once we port to redux, we should be able to connect to the store, and have the App component simply take ```errors``` as a prop. It'll make it easier to test since you won't have to instantiate a store, simply just ```<App errors={['You must login']}/>```